p_sets/3/generate.rb

```diff
@@ -0,0 +1,26 @@
+class RandomArrayGenerator
+  EXPONENT = 10**3
+  attr_reader :length, :seed
+
+  def initialize(length, seed = nil)
+    @length = length
+    @seed = seed
+  end
+
+  def array
+    Array.new(length) { (rand * EXPONENT).ceil }
+  end
+
+  def seed_rand
+    srand seed
+  end
+end
+
+if ARGV.count > 2 || ARGV.empty?
```

Comment:
FWIW, this doesn't catch the case of there being only one argument supplied (which will cause an error in a couple of lines).